src/main/java/com/hazelcast/jdbc/JdbcConnection.java

```
@@ -0,0 +1,352 @@
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(String sql) throws SQLException {
+        checkClosed();
+        return new HazelcastJdbcPreparedStatement(sql, client);
+    }
+
+    @Override
+    public CallableStatement prepareCall(String sql) throws SQLException {
+        checkClosed();
+        throw JdbcUtils.unsupported("CallableStatement not supported");
+    }
+
+    @Override
+    public String nativeSQL(String sql) throws SQLException {
+        checkClosed();
+        return sql;
+    }
+
+    @Override
+    public void setAutoCommit(boolean autoCommit) throws SQLException {
+        checkClosed();
+    }
+
+    @Override
+    public boolean getAutoCommit() throws SQLException {
+        checkClosed();
+        return false;
+    }
+
+    @Override
+    public void commit() throws SQLException {
+        checkClosed();
+    }
+
+    @Override
+    public void rollback() throws SQLException {
+        checkClosed();
+    }
+
+    @Override
+    public void close() throws SQLException {
+        if (!isClosed()) {
+            client.shutdown();
+            closed = true;
+        }
+    }
+
+    @Override
+    public boolean isClosed() throws SQLException {
+        return closed;
+    }
+
+    @Override
+    public DatabaseMetaData getMetaData() throws SQLException {
+        checkClosed();
+        return null;
+    }
+
+    @Override
+    public void setReadOnly(boolean readOnly) throws SQLException {
+        checkClosed();
+    }
+
+    @Override
+    public boolean isReadOnly() throws SQLException {
+        checkClosed();
+        return false;
+    }
+
+    @Override
+    public void setCatalog(String catalog) throws SQLException {
+        checkClosed();
+    }
+
+    @Override
+    public String getCatalog() throws SQLException {
+        checkClosed();
+        return null;
+    }
+
+    @Override
+    public void setTransactionIsolation(int level) throws SQLException {
+        checkClosed();
```

As per JavaDoc, the passed value should be validated.

---

The comment above is not addressed - the method should throw an exception if the passed value is not one of the predefined constants.